cmake/Modules/CocosConfigDefine.cmake

```diff
@@ -74,8 +74,9 @@ define_property(TARGET
 # check c++ standard
 set(CMAKE_C_STANDARD 99)
 set(CMAKE_C_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
C++17 is not supported by some older compilers.

Reply:
This was discussed on the [forum](https://discuss.cocos2d-x.org/t/v4-would-be-a-good-time-to-move-things-forward/47962) and the consensus was that things are moved forward for the v4 branch.  If a user moved to v4 they would need to update their toolchain to the minimum required (as yet undefined).  If they cannot update their toolchain then they can stay at v3.

Reply:
We haven't tested C++14, not even C++17. Metal support is already a big change, i think it is better not add other things to make it not stable. We can update it after it is stable. And developers can change it if needed, it is not difficult.

Reply:
That seems to be at odds with our discussion on the forum where I thought you were in favour of going to C++14/17 and moving classes out of the `experimental` namespace?

Reply:
Yep, i agree to move experimental, but i don't think update to C++14/17 is a good idea.

Reply:
That's fine.  I will update the commit this evening.